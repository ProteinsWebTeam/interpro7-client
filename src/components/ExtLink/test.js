// @flow
/* eslint react/jsx-key: 0 */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { BaseLink, PMCLink, DOILink, GoLink, UniProtLink } from '.';

const renderer = new ShallowRenderer();

describe('External links', () => {
  describe('<BaseLink />', () => {
    test('should render simple links', () => {
      const examples = new Set([
        {
          pattern: 'http://www.example.com/{id}',
          id: 'some_id',
        },
        {
          pattern: 'https://example.com/{id}/something',
          id: 999,
        },
        {
          pattern: 'example.com/id:{id}',
          id: 'Identification',
        },
      ]);
      const children = new Set([
        <div>link</div>,
        <div>
          div<span>span</span>
        </div>,
        'link',
      ]);

      for (const { pattern, id } of examples) {
        for (const child of children) {
          renderer.render(
            <BaseLink id={id} pattern={pattern} target="_blank">
              {child}
            </BaseLink>,
          );
          expect(renderer.getRenderOutput()).toMatchSnapshot();
        }
      }
    });
  });

  describe('<PMCLink />', () => {
    test('should render simple links', () => {
      renderer.render(<PMCLink id={12345} />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });

  describe('<DOILink />', () => {
    test('should render simple links', () => {
      renderer.render(<DOILink id="1.2.a.b" />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });

  describe('<GoLink />', () => {
    test('should render simple links', () => {
      for (const id of ['GO_0003676', 'GO:0003676']) {
        renderer.render(<GoLink id={id} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      }
    });
  });

  describe('<UniProtLink />', () => {
    test('should render simple links', () => {
      renderer.render(<UniProtLink id="A0A023GPI8" />);
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
});
