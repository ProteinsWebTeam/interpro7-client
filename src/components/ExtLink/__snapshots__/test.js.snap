// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`External links <DOILink /> should render simple links 1`] = `
<BaseLink
  id="1.2.a.b"
  pattern="{id}"
  target="_blank"
>
  1.2.a.b
</BaseLink>
`;

exports[`External links <GoLink /> should render simple links 1`] = `
<BaseLink
  id="GO:0003676"
  pattern="https://www.ebi.ac.uk/QuickGO/GTerm?id={id}"
  target="_blank"
>
  GO:0003676
</BaseLink>
`;

exports[`External links <GoLink /> should render simple links 2`] = `
<BaseLink
  id="GO:0003676"
  pattern="https://www.ebi.ac.uk/QuickGO/GTerm?id={id}"
  target="_blank"
>
  GO:0003676
</BaseLink>
`;

exports[`External links <PDBeLink /> should render simple links 1`] = `
<BaseLink
  id="101m"
  pattern="https://www.ebi.ac.uk/pdbe/entry/pdb/{id}"
  target="_blank"
>
  101m
</BaseLink>
`;

exports[`External links <PMCLink /> should render simple links 1`] = `
<BaseLink
  id="12345"
  pattern="https://europepmc.org/abstract/MED/{id}"
  target="_blank"
>
  PUB12345
</BaseLink>
`;

exports[`External links <TaxLink /> should render simple links 1`] = `
<BaseLink
  id={12345}
  pattern="https://www.ebi.ac.uk/ena/data/view/Taxon:{id}"
  target="_blank"
>
  TaxID 12345
</BaseLink>
`;

exports[`External links <UniProtLink /> should render simple links 1`] = `
<BaseLink
  id="A0A023GPI8"
  pattern="http://www.uniprot.org/uniprot/{id}"
  target="_blank"
>
  A0A023GPI8
</BaseLink>
`;
